### PR TITLE
improved Take Screenshot to be able not to log and to return a Screen…

### PR DESCRIPTION
**Pull request by Snooz82**
Take Screenshot got two more arguments:
- return_as: defines wether the keyword returns a path or image
- log_screenshot: defines wether the screenshot is logged to log.html or not

Also changes positional_or_named arguments to named_only arguments

### CODE REVIEW

File: Browser/keywords/browser_control.py [hunk 1]
```diff
@@ -125,84 +151,147 @@ def take_screenshot(
...
-            return self._log_image_link(file_path)
-
-    def _log_image_link(self, file_path: str) -> str:
+            screenshot_path_str = response.body
+            screenshot_path = Path(screenshot_path_str)
+            if (log_screenshot and self._is_embed(filename)) or return_as in (
+                ScreenshotReturnType.bytes,
+                ScreenshotReturnType.base64,
+            ):
+                with screenshot_path.open("rb") as png_file:
```
Comment (aaltat) [inline Browser/keywords/browser_control.py:202]: read_bytes might be handy too: https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_bytes

File: Browser/keywords/browser_control.py [hunk 2]
```diff
@@ -125,84 +151,147 @@ def take_screenshot(
...
+            if log_screenshot and not self._is_embed(filename):
+                self._log_image_link(screenshot_path_str)
+
+            self._unlink_screenshot(filename, return_as, screenshot_path)
+            if return_as is ScreenshotReturnType.path_string:
+                return "EMBED" if self._is_embed(filename) else screenshot_path_str
+            if return_as is ScreenshotReturnType.path:
+                return screenshot_path
+            if return_as is ScreenshotReturnType.bytes:
+                return screenshot_bytes
```
Comment (aaltat) [inline Browser/keywords/browser_control.py:218]: Does this have a test?

File: Browser/keywords/browser_control.py [hunk 3]
```diff
@@ -125,84 +151,147 @@ def take_screenshot(
...
+            if return_as is ScreenshotReturnType.base64:
+                return base64_screenshot.decode()
```
Comment (aaltat) [inline Browser/keywords/browser_control.py:220]: Is there test for this one?

**Comment by aaltat** [comment]
You could update with main and see does the code complexity calculation nag anything?

File: Browser/keywords/browser_control.py (hunk 2 above)
Comment (Snooz82) [reply Browser/keywords/browser_control.py:218]: not yet.

**Pull request merged**